### PR TITLE
Rane Wallin - WebAPI Challenge III

### PR DESCRIPTION
**Pull request by RaneWallin**

### CODE REVIEW

**Comment by RaneWallin** [comment]
MVP  Complete

**Comment by RaneWallin** [comment]
MVP Complete for real

File: index.js [hunk 1]
```diff
@@ -1 +1,6 @@
 // code away!
+const server = require('./server');
+
```
Comment (GannonDetroit) [inline index.js]: good job

File: middleware/logger.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = (req, res, next) => {
```
Comment (GannonDetroit) [inline middleware/logger.js]: good job on the logger

File: middleware/validateBody.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module.exports = (req, res, next) => {
```
Comment (GannonDetroit) [inline middleware/validateBody.js]: good job on the body validator

File: middleware/validatePost.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module.exports = (req, res, next) => {
```
Comment (GannonDetroit) [inline middleware/validatePost.js]: good job on the post validator

File: middleware/validatePostId.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const postDb = require("../posts/postDb");
+
+module.exports = async (req, res, next) => {
```
Comment (GannonDetroit) [inline middleware/validatePostId.js]: good job on the post by id validator

File: middleware/validateUser.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+module.exports = (req, res, next) => {
```
Comment (GannonDetroit) [inline middleware/validateUser.js]: good job on the user validator

File: middleware/validateUserId.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+const userDb = require('../users/userDb');
+
```
Comment (GannonDetroit) [inline middleware/validateUserId.js]: good job on the user by id  validator

File: posts/postRouter.js [hunk 8]
```diff
@@ -1,27 +1,53 @@
-const express = 'express';
+const express = require("express");
```
Comment (GannonDetroit) [inline posts/postRouter.js]: good job on your Post router and the CRUD endpoints

File: server.js [hunk 9]
```diff
@@ -1,15 +1,19 @@
-const express = 'express';
+const express = require('express');
+const userRouter = require('./users/userRouter');
+const postRouter  = require('./posts/postRouter');
```
Comment (GannonDetroit) [inline server.js]: good job on your server and brining in your routes and middleware

File: users/userRouter.js [hunk 10]
```diff
@@ -1,47 +1,92 @@
-const express = 'express';
+const express = require("express");
+const validateUser = require("../middleware/validateUser");
+const validateBody = require("../middleware/validateBody");
+const validateUserId = require("../middleware/validateUserId");
+const validatePost = require("../middleware/validatePost");
+const userDb = require("../users/userDb");
+const postDb = require("../posts/postDb");
```
Comment (GannonDetroit) [inline users/userRouter.js]: good job on your User router and the CRUD endpoints

**Pull request merged**